dowhy/gcm/falsify.py

```diff
@@ -480,25 +488,25 @@ def __repr__(self):
         for m in self.suggestions:
             suggestion_repr[FALSIFY_METHODS[m]] = [
                 f"Remove edge {node[1]} --> {node[0]}"
-                for (node, r) in self.suggestions[m][FALSIFY_P_VALUES].items()
+                for (node, r) in self.suggestions[m][FalsifyConst.P_VALUES].items()
                 if r[1]
             ]
         return _generate_table(validation_repr, suggestion_repr)
 
     def _can_evaluate(self):
         can_evaluate = True
-        for m in (validate_lmc, validate_parental_dsep):
-            if m not in self.methods:
-                warnings.warn(f"Method {m.__name__} not in methods and thus graph cannot be evaluated!")
+        for m in (validate_lmc, validate_tpa):
+            if m.__name__ not in [n.__name__ for n in self.methods]:
                 can_evaluate = False
+
         return can_evaluate
 
 
 def falsify_graph(
     causal_graph: DirectedGraph,
-    data: pd.DataFrame,
-    methods: Union[Callable, Tuple[Callable, ...]] = (validate_lmc, validate_parental_dsep),
-    suggestion_methods: Union[Callable, Tuple[Callable, ...]] = (validate_causal_minimality),
+    data: Optional[pd.DataFrame] = None,
+    methods: Optional[Union[Callable, Tuple[Callable, ...]]] = None,
+    suggestion_methods: Optional[Union[Callable, Tuple[Callable, ...]]] = None,
```

Comment:
Similar comments as for `validate_graph`.

Reply:
I made data non-optional. Defining the default methods similar to how we do it for `validate_graph` now is a bit tricky though because of `causal_graph_reference` for `validate_tpa` and the p-value cache that is initialized after calling the function.